trillian/migrillian/core/controller.go

```diff
@@ -214,8 +214,9 @@ func (c *Controller) runWithRestarts(ctx context.Context) error {
 	}
 	for err != nil && ctx.Err() == nil {
 		klog.Errorf("%s: Controller.Run: %v", c.label, err)
-		sleepRandom(ctx, 0, c.opts.StartDelay)
-		err = c.Run(ctx)
+		if slerr := sleepRandom(ctx, 0, c.opts.StartDelay); slerr == nil {
```

Comment:
It might well be fine in practice, but just wanted to point out that this will now have different behaviour than it previously had; `c.Run` happened regardless of whether or not `sleepRandom` returned an error (i.e with the old behaviour, even if `sleepRandom` turns out to currently be hopelessly broken, we'd still successfully complete a controller run if it's possible to do so.)

Reply:
That's true, but I went down the stack and the sleep function is a wrapper around `clock.SleepContext` which says:

```
// SleepContext sleeps for at least the specified duration. Returns ctx.Err()
// iff the context is done before the deadline.
```

So the only situation this errors is where the context is cancelled, in which case we shouldn't be embarking on a new adventure anyway. Happy to change the logic back to starting a doomed adventure if you prefer, though if it makes any system calls with that context, or a derived context, it will fail after some cycles anyway.

Reply:
Yeah, I don't feel super strongly about it, just wanted to flag it in case it was unintended.